File: Codigos estudiantes por lenguaje/Java/Bryann Valderrama/BitWise/codigoGray.java
public class Main {

    static String binarytoGray(String binary) {
        String gray = "";
        gray += binary.charAt(0);
        for (int i = 1; i < binary.length(); i++) {
            gray += xor_c(binary.charAt(i - 1), binary.charAt(i));
        }
        return gray;
    }

    static char xor_c(char a, char b) {
        return a == b ? '0' : '1';
    }

    static String graytobinary(String gray) {
        String binary = "";
        binary += gray.charAt(0);
        for (int i = 1; i < gray.length(); i++) {
            if (gray.charAt(i) == '0') {
                binary += binary.charAt(i - 1);
            } else {
                binary += flip(binary.charAt(i - 1));
            }
        }
        return binary;
    }

    static char flip(char c) {
        return c == '0' ? '1' : '0';
    }

    public static void main(String[] args) {
        String b1 = "01001";
        String res = binarytoGray(b1);
        System.out.println(b1 + " -> " + res);
        String res2 = graytobinary(res);
        System.out.println(res + " -> " + res2);
    }
}

File: Codigos estudiantes por lenguaje/PY/Bryann Valderrama/BitWise/codigoGray.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def binarytoGray(binary):
    gray = ''
    gray += binary[0]
    for i in range(1, len(binary)):
        gray += xor_c(binary[i-1], binary[i])
    return gray


def xor_c(a, b):
    return '0' if a == b else '1'


def flip(c):
    return '1' if c == '0' else '0'


def graytoBinary(gray):
    binary = ''
    binary += gray[0]
    for i in range(1, len(gray)):
        if gray[i] == '0':
            binary += binary[i-1]
        else:
            binary += flip(binary[i-1])
    return binary


b1 = '01001'
res = binarytoGray(b1)
wr(f'{b1} -> {res}\n')
res2 = graytoBinary(res)
wr(f'{res} -> {res2}')
